src/page-modules/departures/client/geocoder/index.ts

```diff
@@ -0,0 +1,24 @@
+import { swrFetcher } from '@atb/modules/api-browser';
+import { GeocoderFeature } from '../../types';
+import useSWR from 'swr';
+import useDebounce from '@atb/utils/use-debounce';
+
+export type AutocompleteApiReturnType = GeocoderFeature[];
+export type ReverseApiReturnType = GeocoderFeature | undefined;
+
+export function useAutocomplete(q: string) {
+  const debouncedQuery = useDebounce(q, 500);
+
+  return useSWR<AutocompleteApiReturnType>(
+    `/api/departures/autocomplete?q=${debouncedQuery}`,
+    swrFetcher,
+  );
+}
+
+export async function reverse(coords: GeolocationCoordinates) {
+  const result = await fetch(
```

Comment:
Should we call the API differently from the client, @mikaelbr? Didn't want to use `useSWR` in this case, just a regular fetch.

Reply:
I think for GET requests it's good to use `useSWR` it has SSR support, does caching and can do some polling/mutations if we want. And it's easily composable with other hooks.

Reply:
Tried using it to begin with, but couldn't quite achieve what I wanted. The flow became too asynchronous for me and I felt I didn't have too much control of when I got a position. 

Had something like this. 
```tsx
export default function GeolocationButton({
  onGeolocate,
  className,
}: GeolocationButtonProps) {
  const { position, getPosition, isLoading, isUnavailable } =
    useGeolocation();
  const { data} = useReverse(position?.coords);

  useEffect(() => {
    if (data) {
      onGeolocate(data);
    }
  }, [data]);

  if (isUnavailable) return null;

  return ...
}

function useGeolocation() {
  const [position, setPosition] = useState<GeolocationPosition | null>(null);
  const [error, setError] = useState<GeolocationPositionError | null>(null);
  const [isLoading, setIsLoading] = useState(false);
  const [isUnavailable, setIsUnavailable] = useState(false);

  useEffect(() => {
    if (!navigator.geolocation) setIsUnavailable(true);
  }, []);

  const getPosition = () => {
    setIsLoading(true);

    navigator.geolocation.getCurrentPosition(
      (position) => {
        setIsLoading(false);
        setPosition(position);
      },
      (error) => {
        setIsLoading(false);
        setError(error);
      },
      { enableHighAccuracy: true, timeout: 10000 },
    );
  };

  return {
    position,
    setPosition,
    getPosition,
    getPositionAsync,
    error,
    isLoading,
    isUnavailable,
  };
}
```

Reply:
Yep. https://swr.vercel.app/docs/conditional-fetching

Reply:
Right. I can try that approach some more. The problem I had (one of them at least) was that if the geolocation had been called once, it wouldn't update the data and thus not call `onGeolocate`. Couldn't figure out an elegant way to trigger the call of `onGeolocate` whilst assuring that the fetching of data has succeeded (if it's cached it might not be too big of an issue).

Reply:
Do you have any tips/recommendation for how to model the data fetching and updating the selected item in the search box when using `useSWR` and relying on state changes?

Reply:
Decided not to useSWR here. Resolving comment thread